### ISSUE TITLE
SortBuilder toString throws an exception

### ISSUE THREAD

**Issue by OdinZheng** [elastic/elasticsearch#20853]
**Elasticsearch version**: 2.4.1

**Plugins installed**: []

**JVM version**: 1.8.0_31

**OS version**: Windows 10

**Description of the problem including expected versus actual behavior**:

Trying to print a SortBuilder throws an exception. Expected behaviour is that toString returns a string.

This did work on 2.3.3

**Steps to reproduce**:
 1. Run this code:

````
import org.elasticsearch.search.sort.SortBuilder
import org.elasticsearch.search.sort.SortBuilders
import org.elasticsearch.search.sort.SortOrder

SortBuilder sort = SortBuilders.fieldSort('foo').order(SortOrder.DESC)

println sort
````

**Provide logs (if relevant)**:

Error message is:

````
Exception thrown
ElasticsearchException[Failed to build query]; nested: JsonGenerationException[Can not write a field name, expecting a value];
	at org.elasticsearch.search.sort.SortBuilder.toString(SortBuilder.java:40)
	at ConsoleScript2.run(ConsoleScript2:7)
Caused by: com.fasterxml.jackson.core.JsonGenerationException: Can not write a field name, expecting a value
	at com.fasterxml.jackson.core.JsonGenerator._reportError(JsonGenerator.java:1886)
	at com.fasterxml.jackson.core.json.UTF8JsonGenerator._writePPFieldName(UTF8JsonGenerator.java:369)
	at com.fasterxml.jackson.core.json.UTF8JsonGenerator.writeFieldName(UTF8JsonGenerator.java:180)
	at org.elasticsearch.common.xcontent.json.JsonXContentGenerator.writeFieldName(JsonXContentGenerator.java:147)
	at org.elasticsearch.common.xcontent.XContentBuilder.field(XContentBuilder.java:288)
	at org.elasticsearch.common.xcontent.XContentBuilder.field(XContentBuilder.java:270)
	at org.elasticsearch.common.xcontent.XContentBuilder.startObject(XContentBuilder.java:157)
	at org.elasticsearch.search.sort.FieldSortBuilder.toXContent(FieldSortBuilder.java:134)
	at org.elasticsearch.search.sort.SortBuilder.toString(SortBuilder.java:37)
	... 1 more
````

**Comment by OdinZheng** [comment]
@benatwork99 FYI

**Comment by s1monw** [comment]
this is fixed in 5.0 - we only port critical bugfixes to 2.4, that said I don't think we will fix this one

**Comment by juazugas** [comment]
@s1monw as you said, in 2.4.7-SNAPSHOT is still ocurring this issue. I know is closed and is not critical but I've prepared a pull request to solve the problem in the 2.4 branch. Would it be of any interest?
